### PR TITLE
Add syntax sugar to singleton map creation

### PR DESCRIPTION
**Pull request by v1ctor**
Hi!
I have added another syntax sugar to simplify singleton map creation.
for instance:
```java
Map<Integer, Integer> map = HashMap.of(1, 2);
```
instead of
```java
Map<Integer, Integer> map = HashMap.empty().put(1, 2);
```
or
```java
Map<Integer, Integer> map = HashMap.of(Tuple.of(1, 2));
```

P.S. I'm sorry if I'm annoying :)

### CODE REVIEW

File: src/main/java/javaslang/collection/TreeMap.java [hunk 1]
```diff
@@ -99,6 +105,19 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
...
+     *
+     * @param key A singleton map key.
+     * @param value A singleton map value.
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given entry
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(K key, V value) {
+        return of((Comparator<? super K> & Serializable) K::compareTo, Tuple.of(key, value));
+    }
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/TreeMap.java]: Hi, please move this method behind the next method (`of(Comparator, Tuple2)`) and add an additional method

```java
public static <K, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K key, V value) {
    ...
}
```

File: src/main/java/javaslang/collection/LinkedHashMap.java [hunk 2]
```diff
@@ -78,6 +84,21 @@ public LinkedHashMap(Queue<Tuple2<K, V>> list, HashMap<K, V> map) {
...
+     * @param key A singleton map key.
+     * @param value A singleton map value.
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given entry
+     */
+    public static <K, V> LinkedHashMap<K, V> of(K key, V value) {
+        final HashMap<K, V> map = HashMap.of(key, value);
+        final Queue<Tuple2<K, V>> list = Queue.of(Tuple.of(key, value));
+        return list.isEmpty() ? empty() : new LinkedHashMap<>(list, map);
```
Comment (ruslansennov) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: list.isEmpty()? :smile:

Comment (v1ctor) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: It's the same like in method above.

**Comment by danieldietrich** [comment]
> P.S. I'm sorry if I'm annoying :)

It isn't annoying! Your feedback improves the usability!

File: src/main/java/javaslang/collection/LinkedHashMap.java (hunk 2 above)
Comment (danieldietrich) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: whooops :-)

Comment (danieldietrich) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: please correct it...

Comment (v1ctor) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: :-D
Yes, I'll do that.

Comment (ruslansennov) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: whooooooooops :)

Comment (danieldietrich) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: hehe

File: src/main/java/javaslang/collection/LinkedHashMap.java [hunk 3]
```diff
@@ -74,7 +80,22 @@ public LinkedHashMap(Queue<Tuple2<K, V>> list, HashMap<K, V> map) {
     public static <K, V> LinkedHashMap<K, V> of(Tuple2<? extends K, ? extends V> entry) {
         final HashMap<K, V> map = HashMap.of(entry);
         final Queue<Tuple2<K, V>> list = Queue.of((Tuple2<K, V>) entry);
-        return list.isEmpty() ? empty() : new LinkedHashMap<>(list, map);
+        return new LinkedHashMap<>(list, map);
+    }
+
+    /**
+     * Returns a singleton {@code HashMap}, i.e. a {@code HashMap} of one element.
```
Comment (ruslansennov) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: {@code LinkedHashMap}

File: src/main/java/javaslang/collection/TreeMap.java [hunk 4]
```diff
@@ -114,6 +120,34 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
     }
 
     /**
+     * Returns a singleton {@code HashMap}, i.e. a {@code HashMap} of one element.
```
Comment (ruslansennov) [inline src/main/java/javaslang/collection/TreeMap.java]: {@code TreeMap}

File: src/main/java/javaslang/collection/TreeMap.java [hunk 5]
```diff
@@ -114,6 +120,34 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
...
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given entry
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(K key, V value) {
+        return of((Comparator<? super K> & Serializable) K::compareTo, key, value);
+    }
+
+    /**
+     * Returns a singleton {@code HashMap}, i.e. a {@code HashMap} of one element.
```
Comment (ruslansennov) [inline src/main/java/javaslang/collection/TreeMap.java]: {@code TreeMap}

Comment (v1ctor) [inline src/main/java/javaslang/collection/TreeMap.java]: Yes, you're right. I have fixed it.

**Comment by danieldietrich** [comment]
Well done, many thanks!!

**Pull request merged**